app/src/main/java/org/keynote/godtools/android/fragment/LanguagesFragment.java

```diff
@@ -120,6 +145,75 @@ public void onDestroyView() {
         super.onDestroyView();
     }
 
+    @Override
+    public void onSaveInstanceState(@NonNull Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putBoolean(KEY_IS_SEARCH_OPEN, mSearchItem.isActionViewExpanded());
+        outState.putString(KEY_SEARCH, mQuery);
+    }
+
+    //region Initialize Menu
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        super.onCreateOptionsMenu(menu, inflater);
+
+        inflater.inflate(R.menu.fragment_language_search, menu);
+        setSearchView(menu);
+
+    }
+
+    private void setSearchView(Menu menu) {
+        // Configuring the SearchView
+        mSearchItem = menu.findItem(R.id.app_bar_language_search);
+        SearchView mSearchView = (SearchView) mSearchItem.getActionView();
+        mSearchView.setQueryHint(getString(R.string.label_language_search));
+        if (mIsSearchViewOpen) {
+            mSearchItem.expandActionView();
+            if (mQuery != null) {
+                mSearchView.setQuery(mQuery, false);
+            }
+        }
+
+        // Will listen for search event and trigger
+        mSearchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                mQuery = query;
+                updateLanguagesList();
+                return false;
+            }
+
+            @Override
+            public boolean onQueryTextChange(String newText) {
+                mQuery = newText;
+                updateLanguagesList();
+                return false;
+            }
+        });
+    }
+
+    private List<Language> updateLanguageWithSearch(String query) {
+        List<Language> queryList = new ArrayList<>();
+        // Iterate through list and create new list for adapter
+        if (mLanguages != null) {
+            for (Language language : mLanguages) {
```

Comment:
if you wanted to use a stream I believe this is how it would be written:
```
return Stream.of(mLanguages)
    .filter(l -> l.getDisplayName().toLowerCase().contains(query.toLowerCase()))
    .toList()
```

Reply:
This is what I used and it didn't  work for me.  It keep giving me an empty list.

Reply:
OK well it is working now. Java is moody sometimes.